specification/cognitiveservices/data-plane/WebSearch/v1.0/WebSearch.json

```diff
@@ -0,0 +1,1235 @@
+            "in": "query",
+            "description": "The market where the results come from. Typically, mkt is the country where the user is making the request from. However, it could be a different country if the user is not located in a country where Bing delivers results. The market must be in the form <language code>-<country code>. For example, en-US. The string is case insensitive. If known, you are encouraged to always specify the market. Specifying the market helps Bing route the request and return an appropriate and optimal response. If you specify a market that is not listed in Market Codes, Bing uses a best fit market code based on an internal mapping that is subject to change. This parameter and the cc query parameter are mutually exclusive—do not specify both.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "offset",
+            "x-ms-client-name": "Offset",
+            "in": "query",
+            "description": "The zero-based offset that indicates the number of search results to skip before returning results. The default is 0. The offset should be less than (totalEstimatedMatches - count). Use this parameter along with the count parameter to page results. For example, if your user interface displays 10 search results per page, set count to 10 and offset to 0 to get the first page of results. For each subsequent page, increment offset by 10 (for example, 0, 10, 20). it is possible for multiple pages to include some overlap in results.",
+            "required": false,
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "name": "promote",
+            "x-ms-client-name": "Promote",
+            "in": "query",
+            "description": "A comma-delimited list of answers that you want the response to include regardless of their ranking. For example, if you set answerCount to 2 so Bing returns the top two ranked answers, but you also want the response to include news, you'd set promote to news. If the top ranked answers are webpages, images, videos, and relatedSearches, the response includes webpages and images because news is not a ranked answer. But if you set promote to video, Bing would promote the video answer into the response and return webpages, images, and videos. The answers that you want to promote do not count against the answerCount limit. For example, if the ranked answers are news, images, and videos, and you set answerCount to 1 and promote to news, the response contains news and images. Or, if the ranked answers are videos, images, and news, the response contains videos and news. Possible values are Computation, Images, News, RelatedSearches, SpellSuggestions, TimeZone, Videos, Webpages. Use only if you specify answerCount.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "q",
+            "x-ms-client-name": "Query",
```

Comment:
It occurred to me that you might want to have this parameter at the top, since it's the most significant.  (it's `required`, so it should be listed before the others in generated code anyway)